### PR TITLE
Duck

### PR DESCRIPTION
**Pull request by gbtami**

### CODE REVIEW

**Review by Fulmene — CHANGES_REQUESTED** [review]

File: client/analysisCtrl.ts [hunk 1]
```diff
@@ -793,13 +798,11 @@ export class AnalysisController extends GameController {
         return `${event}\n${site}\n${date}\n${white}\n${black}\n${result}\n${variant}\n${fen}\n${setup}\n\n${moveText} *\n`;
     }
 
-    doSendMove = (orig: cg.Orig, dest: cg.Key, promo: string) => {
-        const move = cg2uci(orig + dest + promo);
```
[the remark this answers is not in the document]
Comment (gbtami) [reply client/analysisCtrl.ts:797]: No. In duck chess the moves Fairy expects are not UCI moves. For example "e2e4,e4f6"

File: client/cgCtrl.ts [hunk 2]
```diff
@@ -65,6 +65,7 @@ export abstract class ChessgroundController implements IBoardController {
 
         new ffishModule().then((loadedModule: any) => {
             this.ffish = loadedModule;
+            console.log(this.ffish.variants());
```
[the remark this answers is not in the document]
Comment (gbtami) [reply client/cgCtrl.ts:68]: No :)

File: client/chess.ts [hunk 3]
```diff
@@ -1085,6 +1105,31 @@ export function moveDests(legalMoves: UCIMove[]): cg.Dests {
...
+// Return the duck square (represented by *) from FEN 
+export function duckSquare(fen: string): undefined | cg.Key {
+    const parts = fen.split('/');
+    const rank = parts.findIndex(part => part.includes('*'));
+    if (rank === -1) return undefined;
+    const duckPart = parts[rank];
+    const duckPartFilled = duckPart.replace(/[1-7]/g, m => '_'.repeat(parseInt(m)));
+    const file = duckPartFilled.indexOf('*');
+    return `${'abcdefgh'.charAt(file)}${8-rank}` as cg.Key;
+}
```
[the remark this answers is not in the document]
Comment (gbtami) [reply client/chess.ts]: Done.

File: package.json [hunk 4]
```diff
@@ -15,7 +15,7 @@
   },
   "funding": "patreon.com/pychess",
   "dependencies": {
-    "chessgroundx": "^10.2.2",
+    "chessgroundx": "^10.2.4",
```
[the remark this answers is not in the document]
Comment (gbtami) [reply package.json:18]: I was thinking about it, but was not brave enough to do it :)

File: client/analysisCtrl.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Fulmene) [reply client/analysisCtrl.ts:797]: I mean they're still in the UCI-style square notation: if the board were 10x10 then it expects `e10` and not `e:`.

[the remark this answers is not in the document]
Comment (gbtami) [reply client/analysisCtrl.ts:797]: Yea, the cg2uci conversation is there in both places where do SendMove() called.
In gameCtrl sendmove() and in analysisCtrl makePvMove() as well.

[the remark this answers is not in the document]
Comment (Fulmene) [reply client/analysisCtrl.ts:797]: Sounds good to me.

**Pull request merged**